yt/frontends/amrvac/fields.py

```diff
@@ -29,14 +30,45 @@
     "spherical": ("r", "theta", "phi")
 }
 
+def _velocity(field, data, idir, prefix=None):
+    """Velocity = linear momentum / density"""
+    # This is meant to be used with functools.partial to produce
+    # functions with only 2 arguments (field, data)
+    # idir : int
+    #    the direction index (1, 2 or 3)
+    # prefix : str
+    #    used to generalize to dust fields
+    if prefix is None:
+        prefix = ""
+    moment = data["gas", "%smoment_%d" % (prefix, idir)]
```

Comment:
I am genuinely pleased to see index-by-one for moment fields, although I don't know why.  Index-by-zero for axes always bugs me.